### PR TITLE
Fix mismatched Yul optimizer step lists in docs

### LINKED ISSUE #12081: [Docs] Mismatched Yul optimizer steps lists
Looking at the list of Yul transformations in the documentation, I noticed a few slight differences between the list at https://docs.soliditylang.org/en/latest/yul.html#optimization-step-sequence and the list at https://docs.soliditylang.org/en/latest/internals/optimizer.html#optimizer-steps:

- [ ] In the second list, change FunctionalInliner to ExpressionInliner, just like in the first list.
- [ ] In the second list, change FullFunctionInliner to FullInliner, just like in the first list.
- [ ] Add UnusedFunctionParamterPruner to the first list.

Reported by @acoglio

### PR DESCRIPTION
**Pull request by mau211**
fixes #12081,

Made updates to references reported in pr #12081

### CODE REVIEW

File: docs/internals/optimizer.rst [hunk 1]
```diff
@@ -1109,7 +1109,7 @@ This component can only be used on sources with unique names.
 
 .. _full-function-inliner:
 
-FullFunctionInliner
+FullInliner
 ^^^^^^^^^^^^^^^^^^^
```
Comment (cameel) [inline docs/internals/optimizer.rst]:
```suggestion
FullInliner
^^^^^^^^^^^
```

Comment (mau211) [reply docs/internals/optimizer.rst]: I committed your suggestion via GitHub

File: docs/yul.rst [hunk 2]
```diff
@@ -1263,6 +1263,7 @@ Abbreviation Full name
 ``a``        ``SSATransform``
 ``t``        ``StructuralSimplifier``
 ``u``        ``UnusedPruner``
+``p``        ``UnusedFunctionParamterPruner``
```
Comment (cameel) [inline docs/yul.rst]:
```suggestion
``p``        ``UnusedFunctionParameterPruner``
```

File: docs/internals/optimizer.rst [hunk 3]
```diff
@@ -1084,7 +1084,7 @@ Function Inlining
 
 .. _functional-inliner:
```
Comment (cameel) [inline docs/internals/optimizer.rst]: The references need to be updated too.

File: docs/internals/optimizer.rst [hunk 4]
```diff
@@ -1084,7 +1084,7 @@ Function Inlining
...
 
-FunctionalInliner
+ExpressionInliner
```
Comment (cameel) [inline docs/internals/optimizer.rst:1088]: There's a section called `Optimizer Steps` above. It has a list of these steps and it still has the old names.

**Review by cameel — COMMENTED** [review]
Since this PR fixes #12081, you should add `Fixes #12081` to the PR description. This way github will close the issue automatically when this is merged.

Also, there's no need to copy the whole issue description to the PR. If you really want to quote it, please use proper syntax, so that it does not look like your own text. I.e.

```markdown
> Looking at the list of Yul transformations in the documentation, I noticed a few slight differences between the list at https://docs.soliditylang.org/en/latest/yul.html#optimization-step-sequence and the list at https://docs.soliditylang.org/en/latest/internals/optimizer.html#optimizer-steps:
>
> - [ ] In the second list, change FunctionalInliner to ExpressionInliner, just like in the first list.
> - [ ] In the second list, change FullFunctionInliner to FullInliner, just like in the first list.
> - [ ] Add UnusedFunctionParamterPruner to the first list.
>
> Reported by @acoglio 
```

will be rendered like this:

> Looking at the list of Yul transformations in the documentation, I noticed a few slight differences between the list at https://docs.soliditylang.org/en/latest/yul.html#optimization-step-sequence and the list at https://docs.soliditylang.org/en/latest/internals/optimizer.html#optimizer-steps:
>
> - [ ] In the second list, change FunctionalInliner to ExpressionInliner, just like in the first list.
> - [ ] In the second list, change FullFunctionInliner to FullInliner, just like in the first list.
> - [ ] Add UnusedFunctionParamterPruner to the first list.
>
> Reported by @acoglio

File: docs/internals/optimizer.rst (hunk 4 above)
Comment (mau211) [reply docs/internals/optimizer.rst:1088]: Noted, updated a field for Optimizer steps section

**Review by cameel — APPROVED** [review]
A link to the Full Inliner was still missing and it was still called "Full Function Inliner" in two places in the text but I just pushed a change to fix those. I also squashed your commits (we usually do one commit per logical change and do not leave review fixes as separate commits) and rebased it on `develop`.

In this form the PR seems done do me. Thanks for submitting it!

**Comment by mau211** [comment]
> A link to the Full Inliner was still missing and it was still called "Full Function Inliner" in two places in the text but I just pushed a change to fix those. I also squashed your commits (we usually do one commit per logical change and do not leave review fixes as separate commits) and rebased it on `develop`.
> 
> In this form the PR seems done do me. Thanks for submitting it!

@cameel thank you very much :)

**Comment by mau211** [comment]
@cameel I do not have access to merge it. Can you merge the changes and close the pr. You can also label it "hacktoberfest-accepted"

**Pull request merged**